mesonbuild/compilers/cuda.py

```diff
@@ -46,18 +46,40 @@ def __init__(self, exelist, version, for_machine: MachineChoice,
 
     @classmethod
     def _to_host_flags(cls, flags, phase='compiler'):
-        return list(map(partial(cls._to_host_flag, phase=phase), flags))
+        # since version 6, GCC is *VERY* allergic to passing
+        # '-isystem/usr/include' on the command line, see for instance
+        #
+        #   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70129#c1
+        #
+        # and while nvcc doesn't care about this, the fact that
+        # it happily passes '-isystem/usr/include' to GCC for
+        # compiling host code means we need to filter it.
+        FILTER = frozenset(['-isystem/usr/include']) if phase == 'compiler' else frozenset()
+        prefiltered_flags = filter(lambda x: x not in FILTER, flags)
```

Comment:
We discussed this on IRC too, but if `-isystem` is broken with GCC, maybe we should never emit it inside the Boost dependency when using GCC? It seems wrong that we're filtering a flag that we ourselves emit elsewhere.

Reply:
The problem is, the `-I -> -isystem` rewrite happens at a later phase, when the boost dependency doesn't have control over it anymore. and only `-isystem/usr/include` breaks GCC, not `-I/usr/include`